src/cirq_iqm/iqm_sampler.py

```diff
@@ -180,3 +194,40 @@ def _send_circuits(
                 warnings.warn(f'Failed to abort job: {e}')
             finally:
                 sys.exit()
+
+
+@dataclass
+class ResultMetadata:
+    """Metadata for an IQM execution result.
+
+    Attributes:
+        job_id: A UUID representing the job.
+        calibration_set_id: A UUID representing the calibration set used for this result.
+        request: A RunRequest object representing the request made to run the circuit.
+    """
+
+    job_id: UUID
+    calibration_set_id: Optional[UUID]
+    request: RunRequest
+
+
+class IQMResult(cirq.ResultDict):
+    """Stores the results of a quantum circuit execution on an IQM device.
+
+    Args:
+        params: A cirq.ParamResolver of settings used for this result
+        measurements: A dictionary of measurement keys to measurement results. This is a 2-D array of booleans.
+        records: A dictionary of measurement keys to measurement results, which are 3D arrays of dtype bool.
+        metadata: Metadata for results from IQM circuit execution.
```

Comment:
```suggestion
        params: Parameter resolver used for this circuit, if any.
        measurements: Maps measurement keys to measurement results, which are 2-D arrays of dtype bool.
        records: Maps measurement keys to measurement results, which are 3D arrays of dtype bool.
        metadata: Metadata for the circuit execution results.
```

Reply:
It'd be good to explain what the array dimensions mean too, e.g. using something like ``shape == (repetitions, qubits, whatever...)``